Active hunts UI

AlbanTrincherini opened the pull request:
This PR adds all the active hunts ui, it allows the user to look at all its current challenges by displaying them on a list.  The main ui elements are ActiveHunts.kt and ChallengePreview.kt. I also connected these ui elements to the main navigation bar to allow easy access.

This PR implements user story #47, which is linked to tasks #64, #65, #66, #67

AlbanTrincherini commented:
Note that Guillaume's (and indirectly Marwan's) branches were merged into this one before being finished because I had to get some elements there. This makes CodeClimate unhappy because the files now have some of Guillaume's and Marwan's W.I.P files (maybe the final versions should be merged here or on the master again before proceeding ?).

BoyeGuillaume commented on app/src/main/java/com/github/geohunt/app/ui/components/activehunts/ActiveHunts.kt:
```diff
@@ -0,0 +1,137 @@
+@Composable
+fun EmptyChallengesScreen() {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(text = "No challenges yet...\n" +
+                "Go pick some challenges to begin your hunt!")
```
This is just a note, but for next time, if you could use strings resources in order to ease i18n future rework.

BoyeGuillaume commented on app/src/main/java/com/github/geohunt/app/ui/components/activehunts/ActiveHunts.kt:
```diff
@@ -0,0 +1,137 @@
+        Text(text = "No challenges yet...\n" +
+                "Go pick some challenges to begin your hunt!")
+        Spacer(modifier = Modifier.size(10.dp))
+        Button(onClick = { /*TODO*/ }, colors = ButtonDefaults.buttonColors(backgroundColor = geoHuntRed)) {
```
Take a callback so that we can use the navController in the release and ease the testing of the application

BoyeGuillaume commented on app/src/main/java/com/github/geohunt/app/ui/components/activehunts/ChallengePreview.kt:
```diff
@@ -0,0 +1,146 @@
+}
+
+@Composable
+fun ExpirationTime(expirationDate: LocalDateTime?) {
+    val expires = if (expirationDate != null) "Expires in " else "Expires "
+    val expirationDateFmt = DateFormatUtils.formatRemainingTime(expirationDate).lowercase()
+
+    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+        Icon(Icons.Rounded.CalendarMonth,
+                contentDescription = "calendar_icon")
+
+        Spacer(modifier = Modifier.size(10.dp))
+
+        Text(buildAnnotatedString {
+            withStyle(style = SpanStyle(fontWeight = FontWeight.SemiBold)) {
+                append(expires)
+
+                withStyle(style = SpanStyle(color = geoHuntRed)) {
+                    append(expirationDateFmt)
+                }
+
+                append(" !")
+            }
+
+        })
+    }
+}
```
If you take the last commit on my branch we could merge your expiration time format with mine (as such resolving the comment of @Maeeen)

BoyeGuillaume commented on app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:
```diff
@@ -21,31 +39,117 @@ enum class Route(val titleStringId: Int, val route: String, val icon: Composable
-            Text("Active hunts")
+            val user = Authenticator.authInstance.get().user
+
+            if(user == null) {
+                Text("Something went terribly wrong :/")
+            }
```
I believe this should crash the application because this should be handle above (by @Maeeen)

AlbanTrincherini commented:
I added a callback to navigate when there are no challenges (and tested it) and changed how the image was handled. I also remerged @BoyeGuillaume 's branch to have every file up to date.
There now are some merge conflicts but I would prefer if @BoyeGuillaume or @Maeeen could solve them since they don't come from files I was working on.
Thanks in advance !
